src/main/scala/org/constellation/primitives/EdgeDAO.scala

```diff
@@ -394,8 +395,8 @@ class ThreadSafeSnapshotService(concurrentTipService: ConcurrentTipService) {
     }
     acceptCheckpoint.recoverWith {
       case err =>
-        IO.shift *> dao.metrics.incrementMetricAsync("acceptCheckpoint_failure")
-        IO.raiseError(err) //propagate to upper levels
+        dao.metrics.incrementMetricAsync("acceptCheckpoint_failure")
+            .flatTap(_ => IO.raiseError(err)) //propagate to upper levels
```

Comment:
as above

Reply:
error will be swallowed and not propagated to upper levels